2021/peters2022reproducing/openreview/content.tex

```diff
@@ -0,0 +1,276 @@
+The CelebA\footnote{\url{http://mmlab.ie.cuhk.edu.hk/projects/CelebA.html}, accessed 04-02-22} dataset \citep{liu2015faceattributes} consists of 202.599 images of 10.177 different celebrities, along with a list of attributes depicted in the images. It was not clear from the original paper whether the aligned dataset or the original one was used. Moreover, it was only specified that the images were resized to 224x224, but it was not explained how this was done and whether there were any other preprocessing steps, such as normalisation. After communication with the authors it became clear that the aligned dataset was used, and that the images were to be normalised with 0.5 mean and 0.5 standard deviation. Because the Pytorch dataloader loaded in 38 extra columns that were unnecessary in our research, we manually resized the images to 224x224 with a Pytorch transformation. In order to be able to use the cross-entropy function in a later stage, all -1 values of the binary 'blond' and 'male' variables were mapped to 0. The resulting images were used as data X, the hair colour (blond or not) was used as the target variable Y and the sex variable was used as the sensitive attribute D.
+
+To obtain features from the images, we trained a ResNet-50 model \citep{7780459}, initialised with the pre-trained ImageNet weights, for 10 epochs. The features were then extracted from the second to last layer. The last layer was removed and replaced with a layer consisting of two output nodes to form the classifier. 
+
+\subsubsection{Civil Comments dataset}
+The Civil Comments dataset\footnote{\url{https://www.kaggle.com/c/jigsaw-unintended-bias-in-toxicity-classification/data}, accessed 04-02-22} \citep{borkan2019nuanced} is a text-based dataset consisting of 1.999.514 online comments on various news articles, along with metadata about the commenter and a label indicating whether the comment displays toxicity or not. The Kaggle repository does not provide a test set with labels nor a validation set. This meant that exclusively datapoints from the training set were used in our study. Following \citet{lee2021fair}, we let $X$ be the comment text, $Y$ be the binary toxicity label, and $D$ be whether Christianity is mentioned. The dataset does not include mention-of-Christianity values for each data point and therefore all data points without one were dropped. A total of 235.087 comments remained. These were subsequently split into a training, validation and test set using ratios of $0.8, 0.1, 0.1$ respectively. Additionally, the targets $Y$ and mentions of Christianity $D$ were converted to binary values, where values above or equal to $0.5$ were mapped to $1$ and values below $0.5$ were mapped to $0$. Lastly, the comments $X$ were tokenised using the BERT tokeniser\footnote{\url{https://huggingface.co/docs/transformers/model_doc/bert#transformers.BertTokenizer}, accessed 04-02-22}, with max length set to 512, truncation set to true, and padding set to the max length.
```

Comment:
For some reason, the URL `https://huggingface.co/docs/transformers/model_doc/bert#transformers.BertTokenizer` is breaking the build, whereas if you remove the parts following `#` then it works (as in, URL being `https://huggingface.co/docs/transformers/model_doc/bert` only).